go/vt/tabletserver/messager_engine.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tabletserver
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/youtube/vitess/go/vt/dbconfigs"
+	"github.com/youtube/vitess/go/vt/schema"
+)
+
+// MessagerEngine is the engine for handling messages.
+type MessagerEngine struct {
+	mu       sync.Mutex
+	isOpen   bool
+	managers map[string]*MessageManager
+
+	// TODO(sougou): This depndency should be cleaned up.
+	tsv      *TabletServer
+	connpool *ConnPool
```

Comment:
This is using app user. All conn pools require dba user for killing connections.